libs/libfpgabitstream/src/bitstream_manager.cpp

```diff
@@ -296,6 +299,79 @@ void BitstreamManager::add_output_net_id_to_block(
   block_output_net_ids_[block] = output_net_id;
 }
 
+void BitstreamManager::overwrite_bitstream(const std::string& path,
+                                           const bool& value) {
+  bool bad_format = true;
+  size_t index = path.rfind("[");
+  std::string bit_string = "";
+  if (index != std::string::npos && path[path.size() - 1] == ']') {
+    bit_string = path.substr(index + 1, path.size() - index - 2);
+    bad_format = bit_string.size() == 0;
+    auto iter = bit_string.begin();
+    while (!bad_format && iter != bit_string.end()) {
+      bad_format = !std::isdigit(*iter);
+      iter++;
+    }
+  }
+  if (bad_format) {
+    archfpga_throw(__FILE__, __LINE__,
+                   "overwrite_bitstream bit path '%s' does not match format "
+                   "<full path in the hierarchy of FPGA fabric>[bit index]",
+                   path.c_str());
+  } else {
+    size_t bit = (size_t)(std::stoi(bit_string));
+    StringToken tokenizer(path.substr(0, index));
+    std::vector<std::string> blocks = tokenizer.split(".");
+    std::vector<ConfigBlockId> block_ids;
+    ConfigBlockId block_id = ConfigBlockId::INVALID();
+    size_t found = 0;
+    for (size_t i = 0; i < blocks.size(); i++) {
+      if (i == 0) {
+        block_ids = find_bitstream_manager_top_blocks(*this);
+      } else {
+        block_ids = block_children(block_id);
+      }
+      // Reset
+      block_id = ConfigBlockId::INVALID();
+      // Find the one from the list that match the name
```

Comment:
You can simply use the existing API to find a child block with given name

https://github.com/lnis-uofu/OpenFPGA/blob/4a07a32902c1ce69ea083b9fe60b1910beb58b78/libs/libfpgabitstream/src/bitstream_manager.h#L156-L157

Reply:
It can't fullfil my need to look for top level name and child name at the same time.
Unless find_child_block() is updated to support if block_id == INVALID, then look for top level name that match the child_block_name. 
For now, I will stick with my existing code which work for both top level name and child name searching.

Reply:
This is o.k.